### PR TITLE
dasharo-compatibility/315-network-boot.md: add and update test cases

### PR DESCRIPTION
**Pull request by Pokisiekk**
Signed-off-by: Kamil Pokornicki <kamil.pokornicki@3mdeb.com>

### CODE REVIEW

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md [hunk 1]
```diff
@@ -1,13 +1,105 @@
 # Dasharo Compatibility: Network Boot
 
-## Test cases
+## PXE001.001 Dasharo Network Boot is available
 
-### PXE001.001 iPXE network boot
+**Test description**
+
+This test aims to verify that the `iPXE Network boot` option is available and
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md:7]: In my opinion in the `Test description section` we shouldn't use a direct reference to the user.

So this point should be written as follows:
`This test aims to verify, that the iPXE Network boot is available in the Setup Menu and whether, after selecting this boot option Dasharo Network Boot Menu is displayed.`

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md [hunk 2]
```diff
@@ -1,13 +1,105 @@
...
+whether you are entered to `Dasharo Network Boot Menu` after selecting it.
+
+**Test configuration data**
+
+1. `FIRMWARE` = Dasharo
+
+**Test setup**
+
+1. Proceed with the
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md:16]: `Test setup` stages for all test cases in this suite should contain two points:

1. Proceed with the [Generic test setup: firmware](../../generic-test-setup#firmware).
2. Device should be connected to the Internet by using an Ethernet cable.

As all of these sections should be identical, I think that we should create `common section`.

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md [hunk 3]
```diff
@@ -1,13 +1,105 @@
...
+**Test steps**
+
+1. Power on the DUT.
+1. Hold the `BOOT_MENU_KEY` to enter the UEFI Boot Menu.
+1. Select the `iPXE Network boot` option using the arrow keys and press `Enter`.
+
+**Expected result**
+
+1. The `iPXE Network boot` option is available.
+1. After selecting the `iPXE Network boot` option you should enter to
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md:28]: In my opinion, this point should be written without direct reference to the user.

For example:
1. After selecting the `iPXE Network boot` option, the `Dasharo Network Boot Menu` should be displayed.

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md [hunk 4]
```diff
@@ -1,13 +1,105 @@
...
+    `Dasharo Network Boot Menu`.
+
+## PXE002.001 Dasharo network boot menu boot options order is correct
+
+**Test description**
+
+This test aims to verify that `Dasharo Network Boot Menu` contains all the
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md:35]: In my opinion, this point should be written without direct reference to the user.

For example:

This test aims to verify that `Dasharo Network Boot Menu` contains all of the needed options which are in the correct order.

**Review by sulewskiprzemyslaw — CHANGES_REQUESTED** [review]

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md (hunk 1 above)
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md]: applied

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md (hunk 2 above)
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md:25]: applied

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md (hunk 3 above)
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md]: applied

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md (hunk 4 above)
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md]: applied

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md [hunk 5]
```diff
@@ -1,13 +1,105 @@
...
+
+**Test steps**
+
+1. Power on the DUT.
+1. Hold the `BOOT_MENU_KEY` to enter the UEFI Boot Menu.
+1. Select the `iPXE Network boot` option using the arrow keys and press `Enter`.
+
+**Expected result**
+
+`Dasharo Network Boot Menu` contains all the options and they are organized in
```
[the remark this answers is not in the document]
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md]: changed as above

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md [hunk 6]
```diff
@@ -1,13 +1,105 @@
...
+OS installation
+iPXE Shell
+```
+
+## PXE003.001 Autoboot option is available and works correctly
+
+**Test description**
+
+This test aims to verify that the `Autoboot` option in
+`Dasharo Network Boot Menu` works properly.
```
[the remark this answers is not in the document]
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md]: applied

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md [hunk 7]
```diff
@@ -1,13 +1,105 @@
...
+**Test steps**
+
+1. Power on the DUT.
+1. Hold the `BOOT_MENU_KEY` to enter the UEFI Boot Menu.
+1. Select the `iPXE Network boot` option using the arrow keys and press `Enter`.
+1. Select the `Autoboot` option using the arrow keys and press `Enter`.
+
+**Expected result**
+
+If the server assigned to the `Autoboot` option is available in our network
```
[the remark this answers is not in the document]
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md]: applied

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md [hunk 8]
```diff
@@ -1,13 +1,105 @@
...
+after a while menu should appear.
+
+If the server assigned to the `Autoboot` option isn't available in our network,
```
[the remark this answers is not in the document]
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md]: applied

File: docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md [hunk 9]
```diff
@@ -17,21 +109,84 @@ server.
...
-1. Type in `chain --autofree http://boot.netboot.xyz/` to load a boot menu
-1. Enter the "Live CDs" submenu using the arrow keys and Enter.
-1. Select `Debian` -> `Debian Live 11 (bullseye)` -> `Debian 11 Gnome` and
-    press Enter.
+1. Select the `Dasharo Tools Suite` option using the arrow keys and press
+    `Enter`.
+
+**Expected result**
+
+After a while, the DTS menu should appear.
```
[the remark this answers is not in the document]
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/315-network-boot.md]: applied something similar

**Review by sulewskiprzemyslaw — APPROVED** [review]

**Pull request merged**